irony-cdb.el

```diff
@@ -114,6 +128,57 @@ for files that it cannot handle."
 ;; Functions
 ;;
 
+(defun irony-cdb--choose-closest-path (file paths)
+  "Find the \"best\" path in PATHS matching FILE
+
+If any paths in PATHS is belongs to the same directory
+or a subdirectory of file, we disregard other candidates.
+
+For remaining candidates, \"nearest\" is measured as abs. difference
+in path depth.
+- We prefer deeper paths at level +N to those at level -N.
+- If multiple paths are equally good, we return the last one.
+
+Returns nil if paths isn't a list of at least one element.
+"
+  (when (listp paths)
+    (let ((paths (or
+                  ;; if we find a cdb in cwd or below, don't consider other candidates
```

Comment:
> "or below"? care to explain? to me it looks like we will favor a cdb in the current 
> directory and that's all.

Nope. The test is `(lambda (x) (string-prefix-p (file-name-directory file) x))` and any path
which lives in or under that directory will match, by definition.

> even if irony-cdb-search-directory-list does not specify it as a first element. I think we should > not special case the current dir and just take the first best match from 
> irony-cdb-search-directory-list, which defaults to . for the best match.

The current code already does exactly what you asked me for and passes the test you provided (`children-first` is an adapted version of it). 

Please provide a failing test if you've found a bug.